module/mailnews/newsread.c

```diff
@@ -389,8 +411,137 @@ Bool DateFromSeconds(long seconds, char *str)
    if (t == NULL)
       return False;
 
-   sprintf(str, "%s %s %.2ld, %.4ld %.2ld:%.2ld", 
+   snprintf(str, 50, "%s %s %.2ld, %.4ld %.2ld:%.2ld", 
 	   GetString(hInst, weekdays[t->tm_wday]), GetString(hInst, months[t->tm_mon]), 
 	   t->tm_mday, t->tm_year + 1900, t->tm_hour, t->tm_min);
    return True;
 }
+
+void OnColumnClick(LPNMLISTVIEW pLVInfo)
+{
+    int nSortColumn = pLVInfo->iSubItem;
+    BOOL bSortAscending = TRUE;
+
+    if (nSortColumn == currentSortColumn) {
+        // Toggle sort direction if the same column is clicked again
+        currentSortAscending = !currentSortAscending;
+    } else {
+        // Set ascending order for new column
+        currentSortColumn = nSortColumn;
+        currentSortAscending = TRUE;
+    }
+
+    LPARAM lParamSort = (currentSortAscending ? 1 : -1) * (currentSortColumn + 1);
+
+    // Update header text
+    UpdateColumnHeaders(pLVInfo->hdr.hwndFrom, currentSortColumn, currentSortAscending);
+
+    // Sort list
+    ListView_SortItems(pLVInfo->hdr.hwndFrom, CompareListItems, lParamSort);
+}
+
+/****************************************************************************/
+/*
+ * CompareListItems:  Comparison function for sorting items in a list view
+ *   control. This function is used by `ListView_SortItems` to compare two 
+ *   items and determine their sort order based on the specified column and 
+ *   sort direction. It:
+ * 
+ *   - Determines the sort direction and column index from `lParamSort`.
+ *   - Casts `lParam1` and `lParam2` to `NewsArticle` pointers.
+ *   - Constructs stable comparison keys by appending the article index to 
+ *     the title and poster strings.
+ *   - Compares the items based on the specified column and sort direction:
```

Comment:
Function comments should describe what the function does, not how it works.  As it is this doubles the amount of work for anyone in the future, since any internal change they make also needs to change the comment.  I bet you can guess what the likely outcome of that would be.